improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -230,6 +230,143 @@ def test_exception_without_relative_to_threshold(self):
+        forecast_probability_sum = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        forecast_count = np.array([1000, 1000, 1000, 1000, 1000], dtype=np.float32)
+
+        result = Plugin()._combine_undersampled_bins(
+            obs_count, forecast_probability_sum, forecast_count
+        )
+
+        assert_array_equal(
+            result, [obs_count, forecast_probability_sum, forecast_count]
+        )
+
+    def test_one_undersampled_bin_at_top(self):
+        """Test when the highest probability bin is under-sampled."""
+        obs_count = np.array([0, 250, 500, 750, 100], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 250, 500, 750, 100], dtype=np.float32)
+        forecast_count = np.array([1000, 1000, 1000, 1000, 100], dtype=np.float32)
+
+        expected = np.array(
+            [
+                [0, 250, 500, 850],  # Observation count
+                [0, 250, 500, 850],  # Sum of forecast probability
+                [1000, 1000, 1000, 1100],  # Forecast count
+            ]
+        )
+        result = Plugin()._combine_undersampled_bins(
+            obs_count, forecast_probability_sum, forecast_count
+        )
+
+        assert_array_equal(result, expected)
+
+    def test_one_undersampled_bin_at_bottom(self):
+        """Test when the lowest probability bin is under-sampled."""
+        obs_count = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        forecast_count = np.array([100, 1000, 1000, 1000, 1000], dtype=np.float32)
+
+        expected = np.array(
+            [
+                [250, 500, 750, 1000],  # Observation count
+                [250, 500, 750, 1000],  # Sum of forecast probability
+                [1100, 1000, 1000, 1000],  # Forecast count
+            ]
+        )
+        result = Plugin()._combine_undersampled_bins(
+            obs_count, forecast_probability_sum, forecast_count
+        )
+
+        assert_array_equal(result, expected)
+
+    def test_one_undersampled_bin_lower_neighbour(self):
+        """Test for one under-sampled bin that is combined with its lower
+        neighbour."""
+        obs_count = np.array([0, 250, 50, 1500, 1000], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 250, 50, 1500, 1000], dtype=np.float32)
+        forecast_count = np.array([1000, 1000, 100, 2000, 1000], dtype=np.float32)
+
+        expected = np.array(
+            [
+                [0, 300, 1500, 1000],  # Observation count
+                [0, 300, 1500, 1000],  # Sum of forecast probability
+                [1000, 1100, 2000, 1000],  # Forecast count
+            ]
+        )
+        result = Plugin()._combine_undersampled_bins(
+            obs_count, forecast_probability_sum, forecast_count
+        )
+
+        assert_array_equal(result, expected)
+
+    def test_one_undersampled_bin_upper_neighbour(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6014128)